controllers/dscinitialization/dscinitialization_test.go

```diff
@@ -0,0 +1,233 @@
+			desiredDsci := getNewInstance(applicationName)
+			Expect(k8sClient.Create(ctx, desiredDsci)).Should(Succeed())
+			foundDsci := &dsci.DSCInitialization{}
+			Eventually(dsciExists(applicationName, workingNamespace, foundDsci), timeout, interval).Should(BeTrue())
+
+			// then
+			foundRoleBinding := &authv1.RoleBinding{}
+			Eventually(objectExists(applicationNamespace, applicationNamespace, foundRoleBinding), timeout, interval).Should(BeTrue())
+			Expect(foundRoleBinding.UID).To(Equal(createdRoleBinding.UID))
+			Expect(foundRoleBinding.Subjects).To(BeNil())
+		})
+
+		It("Should not update configmap if it exists", func() {
+			applicationName := "configmap-test"
+
+			// given
+			desiredConfigMap := &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ConfigMap",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      configmapName,
+					Namespace: applicationNamespace,
+				},
+				Data: map[string]string{"namespace": "existing-data"},
+			}
+			Expect(k8sClient.Create(ctx, desiredConfigMap)).Should(Succeed())
+			createdConfigMap := &corev1.ConfigMap{}
+			Eventually(objectExists(configmapName, applicationNamespace, createdConfigMap), timeout, interval).Should(BeTrue())
+
+			// when
+			desiredDsci := getNewInstance(applicationName)
+			Expect(k8sClient.Create(ctx, desiredDsci)).Should(Succeed())
+			foundDsci := &dsci.DSCInitialization{}
+			Eventually(dsciExists(applicationName, workingNamespace, foundDsci), timeout, interval).Should(BeTrue())
+
+			// then
+			foundConfigMap := &corev1.ConfigMap{}
+			Eventually(objectExists(configmapName, applicationNamespace, foundConfigMap), timeout, interval).Should(BeTrue())
+			Expect(foundConfigMap.UID).To(Equal(createdConfigMap.UID))
+			Expect(foundConfigMap.Data).To(Equal(map[string]string{"namespace": "existing-data"}))
+			Expect(foundConfigMap.Data).ToNot(Equal(map[string]string{"namespace": applicationNamespace}))
+		})
+
+		It("Should not update namespace if it exists", func() {
+			applicationName := "configmap-test"
+			anotherNamespace := "test-another-ns"
+
+			// when
+			desiredNamespace := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: anotherNamespace,
+				},
+			}
+			Expect(k8sClient.Create(ctx, desiredNamespace)).Should(Succeed())
+			createdNamespace := &corev1.Namespace{}
+			Eventually(namespaceExists(anotherNamespace, createdNamespace), timeout, interval).Should(BeTrue())
+
+			// given
+			desiredDsci := getNewInstance(applicationName)
+			Expect(k8sClient.Create(ctx, desiredDsci)).Should(Succeed())
+			foundDsci := &dsci.DSCInitialization{}
+			Eventually(dsciExists(applicationName, workingNamespace, foundDsci), timeout, interval).Should(BeTrue())
+
+			//then
+			foundApplicationNamespace := &corev1.Namespace{}
+			Eventually(namespaceExists(anotherNamespace, foundApplicationNamespace), timeout, interval).Should(BeTrue())
+			Expect(foundApplicationNamespace.Name).To(Equal(createdNamespace.Name))
```

Comment:
The general flow is `Given -> When -> Then`, or `Arrange -> Act -> Assert`. This means you first set up test fixture/context/data (that's `Given`), then you exercise your component under test (that's `When` part) and lastly you inspect the outcome (that's `Then`). Changing the order makes it confusing :)